### PR TITLE
KNOX-2295 - UI services improvement on Knox Home page

### PR DESCRIPTION
**Pull request by smolnar82**
## What changes were proposed in this pull request?

Removed the data table when displaying UI services and using the services' logos with Angular's material grid. Long descriptions are shown by hovering over the short descriptions in the footer of each tile.
Services are available when you click on their logos.

## How was this patch tested?

Tested manually; see these screenshots:
Screen Shot 2020-03-15 at 11 05 03 PM

Screen Shot 2020-03-15 at 11 12 25 PM

### CODE REVIEW

**Comment by smolnar82** [comment]
One comment: the Hue logo you are seeing in the above screenshot is replaced with one with white background.

**Comment by smolnar82** [comment]
Thanks, @moresandeep !
Trying to answer your comments:
1. I'll try to find a better CM logo
2. If there is no logo -> a JIRA should be submitted. I tried to add logos for all the actively used services.
3. Hue logo is replaced already (see my comment above)

File: knox-homepage-ui/package.json [hunk 1]
```diff
@@ -10,11 +10,13 @@
   },
   "private": true,
   "dependencies": {
+    "@angular/cdk": "^5.2.5",
```
Comment (risdenk) [inline knox-homepage-ui/package.json]: All the other angular versions is 5.2.0 but this is 5.2.5?

Comment (smolnar82) [reply knox-homepage-ui/package.json]: I haven't tested with `5.2.0` as only `5.2.5` was listed as an officially supported version of Angular's material lib:

Screen Shot 2020-03-16 at 3 57 49 PM

**Comment by smolnar82** [comment]
Regarding @risdenk 's comment on licensing: all service logos are logos of Apache projects except for the CM-[UI|API]. I understand his concerns about it and it's a valid question if we want to copy it here.

**Comment by smolnar82** [comment]
To address the logo stuff I figured out the following approach should work: I'll add a `general_logo.png` (with the text of 'Service Logo comes here') and display it for
- any service that does not have its own logo uploaded by this patch (anyone is free to upload their owns logos going forward as a separate patch)
- CM API/UI

**Comment by smolnar82** [comment]
This is how it looks like now:
Screen Shot 2020-03-16 at 5 09 48 PM

**Comment by risdenk** [comment]
@smolnar82  instead of "service logo comes here" can it instead be the text of the service? It would just be plain text on say a white background. That would probably look better.

**Comment by smolnar82** [comment]
> @smolnar82 instead of "service logo comes here" can it instead be the text of the service? It would just be plain text on say a white background. That would probably look better.

So, we either use a static image (as of now; I'm open to change the text though) or try to find out if the referenced image exists and act differently (e.g. have a text on white background).

I don't think the general logo will appear as much; once someone realizes his logo is not displayed I'd imagine that he uploads one quickly -> the 2nd approach just not worth implementing.

**Comment by smolnar82** [comment]
Newest version: 
Screen Shot 2020-03-16 at 6 16 24 PM

**Pull request merged**